### PR TITLE
Tox integration + Coverage Report 

### PR DESCRIPTION
**Pull request by sidpremkumar**
Adding Tox integration to Review-Rot 
* Coverage set to be 60% for all files in reviewrot/* except mailer.py 
* Coverage report outputs as HTML in directory coverage_report/ 
* Flake8 covers reviewrot/ and test/ with max line lengh = 88 
@pbortlov @Lukkra211

### CODE REVIEW

File: .coveragerc [hunk 1]
```diff
@@ -0,0 +1,10 @@
+[run]
+branch = True
```
Comment (pbortlov) [inline .coveragerc]: with or without spaces?

File: README.md [hunk 2]
```diff
@@ -20,16 +20,12 @@ python setup.py develop
...
 ## Tests
-To run the tests in your virtualenv, execute:
-```shell
-python setup.py test
-```
-
-Alternatively, you can use `tox` or `detox` to run the tests against multiple versions of python:
+You can use `tox` or `detox` to run the tests against multiple versions of python:
 ```shell
 sudo dnf install python-detox
```
Comment (pbortlov) [inline README.md]: is there any version for python 3?

File: README.md [hunk 3]
```diff
@@ -20,16 +20,12 @@ python setup.py develop
...
 detox
 ```
+Test outputs are stored under the `coverage_report` directory as HTML files
```
Comment (pbortlov) [inline README.md]: 'coverage_report/' you can add link to the md file like https://stackoverflow.com/questions/7653483/github-relative-link-in-markdown-file

File: setup.py [hunk 4]
```diff
@@ -3,6 +3,18 @@
...
+
+try:
+    with open('test-requirements.txt', 'rb') as f:
+        tests_require = f.read().decode('utf-8').split('\n')
+except IOError:
+    tests_require = []
+
 setup(name='review-rot',
       version='1.0',
       author='Nirzari Iyer',
```
Comment (pbortlov) [inline setup.py]: Add you, me and @mkosiarc as authors.

File: test/test_gitlabtest.yaml [hunk 5]
```diff
@@ -1,6 +0,0 @@
-token: gf34bn667fgdg567v3d789g543d4d6f876432456
```
Comment (pbortlov) [inline test/test_gitlabtest.yaml]: if this was your token, please change it asap.

File: .coveragerc (hunk 1 above)
Comment (sidpremkumar) [reply .coveragerc]: Removed spaces for consistency

File: README.md (hunk 2 above)
Comment (sidpremkumar) [reply README.md]: tox will automatically run against python 2.7 and 3.6. See in tox.ini `envlist = clean, py27, py36, report, flake8`

File: test/test_gitlabtest.yaml (hunk 5 above)
Comment (sidpremkumar) [reply test/test_gitlabtest.yaml]: This was a fake token that was here before me.

File: README.md (hunk 2 above)
Comment (pbortlov) [reply README.md]: cool

File: README.md [hunk 6]
```diff
@@ -20,16 +20,12 @@ python setup.py develop
...
-python setup.py test
-```
-
-Alternatively, you can use `tox` or `detox` to run the tests against multiple versions of python:
+You can use `tox` or `detox` to run the tests against multiple versions of python:
 ```shell
 sudo dnf install python-detox
 detox
 ```
+Test outputs are stored under the [coverage_report](coverage_report) directory as HTML files
```
Comment (pbortlov) [inline README.md]: Does this link work?

**Comment by amarza-rh** [comment]
1. You don't need a special stage for htmlcov.
2. You may want to generate coverage.html for both supported platforms: `py27` and `py36`.
3. No need to run coverage during the flake8 stage.

This can be structured by having a base `[testenv]` and:
*  `testenv:py27` which runs python2.7 tests, coverage and report and
* `testenv:py36` which does the same for python3.6.

The following patch implements that:

```diff
diff --git a/tox.ini b/tox.ini
index a0cc4a7..e009e0f 100644
--- a/tox.ini
+++ b/tox.ini
@@ -1,5 +1,5 @@
 [tox]
-envlist = clean, py27, py36, report, flake8
+envlist = clean, py27, py36, flake8
 
 [testenv]
 deps =
@@ -7,8 +7,6 @@ deps =
     -r{toxinidir}/test-requirements.txt
     pytest-cov
 sitepackages = False
-commands =
-    pytest --cov --cov-append {posargs}
 
 [testenv:flake8]
 skip_install = true
@@ -25,9 +23,10 @@ deps = coverage
 skip_install = true
 commands = coverage erase
 
-[testenv:report]
-deps = coverage
-skip_install = true
+[testenv:py27]
+commands =
+    pytest --cov --cov-append --cov-report html:htmlcov-py27
+
+[testenv:py36]
 commands =
-    coverage report
-    coverage html
\ No newline at end of file
+    pytest --cov --cov-append --cov-report html:htmlcov-py36
```

File: tox.ini [hunk 7]
```diff
@@ -1,10 +1,33 @@
...
+
+[testenv:flake8]
+skip_install = true
+deps = flake8
+commands =
+    flake8 reviewrot/
+    flake8 test/
+
+[flake8]
+max-line-length=88
```
Comment (amarza-rh) [inline tox.ini]: I'd try to stick to the default line length of 80. You can keep this for now however.

File: README.md (hunk 6 above)
Comment (sidpremkumar) [reply README.md]: Yes

**Pull request merged**